node/forest_libp2p/src/peer_manager.rs

```diff
@@ -200,6 +219,61 @@ impl PeerManager {
         }
         removed
     }
+
+    /// Gets peer operation receiver
+    pub fn peer_ops_rx(&self) -> &Receiver<PeerOperation> {
+        &self.peer_ops_rx
+    }
+
+    /// Bans a peer with an optional duration
+    pub async fn ban_peer(
+        &self,
+        peer: PeerId,
+        reason: impl Into<String>,
+        duration: Option<Duration>,
+    ) {
+        let mut locked = self.peer_ban_list.write().await;
+        locked.insert(peer, duration.and_then(|d| Instant::now().checked_add(d)));
+        if let Err(e) = self
+            .peer_ops_tx
+            .send_async(PeerOperation::Ban(peer, reason.into()))
+            .await
+        {
+            warn!("ban_peer err: {e}");
+        }
+    }
+
+    pub async fn background_task(self: Arc<Self>) -> anyhow::Result<()> {
```

Comment:
Maybe a more descriptive name for this function would help? `background_task` feels a bit too generic.

Reply:
How about `peer_operation_event_loop_task`